src/com/amazon/kinesis/streaming/agent/tailing/AbstractParser.java

```diff
@@ -460,11 +492,21 @@ private R buildRecord(int offset, int length) {
         ByteBuffer data = ByteBuffers.getPartialView(currentBuffer, offset, length);
         ++recordsFromCurrentBuffer;
         Preconditions.checkNotNull(currentBufferFile);
-        R record = buildRecord(currentBufferFile, data, toChannelOffset(offset));
-        totalRecordsParsed.incrementAndGet();
+        R record = null;
+        try {
+            record = buildRecord(currentBufferFile, convertData(data), toChannelOffset(offset));
+            totalRecordsParsed.incrementAndGet();
+        } catch (DataConversionException e) {
```

Comment:
Can we fall back to build record of raw data without processing in the case of getting DataConversionException? That way we won't lose data